src/components/AttachmentCarousel/index.js

```diff
@@ -50,7 +55,11 @@ function AttachmentCarousel(props) {
 
     useEffect(() => {
         const initialState = createInitialState(props);
-        if (initialState) {
+        // dismiss the modal when deleting an attachment during its display in preview.
+        const source = tryResolveUrlFromApiRoot(props.source);
+        if (initialState.page === -1 && _.find(attachments, (a) => a.source === source)) {
```

Comment:
Couldn't we have iterated through the `attachments` only once? We're using `_.findIndex` in `createInitialState` and immediately after that using `_.find`. We should move this to a different util and loop only once.

Reply:
@mananjadhav 
I think we can't because the first loop is through the old state `attachments`, but the second is through the new state that created from `createInitialState.attachments`.

Reply:
I just saw. Thanks.